src/rules/newline-per-chained-call/index.test.ts

```diff
@@ -0,0 +1,283 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import {ESLintUtils} from '@typescript-eslint/utils';
+import {newlinePerChainedCall} from './index';
+
+const ruleTester = new ESLintUtils.RuleTester({
+    parser: '@typescript-eslint/parser',
+    parserOptions: {
+        ecmaFeatures: {
+            jsx: true,
+        },
+    },
+});
+
+ruleTester.run('newline-per-chained-call', newlinePerChainedCall, {
+    valid: [
+        {code: '_\n.chain({})\n.map(foo)\n.filter(bar)\n.value();'},
+        {code: 'a()\n.b()\n.c\n.e'},
+        {code: 'const a = m1.m2(); let b = m1.m2();\nvar c = m1.m2()'},
+        {code: 'const a = m1()\n.m2();'},
+        {code: 'const a = m1();'},
+        {code: 'a().b().c'},
+        {code: 'const foo = a.b.c.e.d'},
+        {code: "a.b.c.e.d = 'foo'"},
+        {code: 'a().b().c()'},
+        {code: 'const a = window\n.location\n.href\n.match(/(^[^#]*)/)[0];'},
+        {code: "const a = window['location']\n.href\n.match(/(^[^#]*)/)[0];"},
+        {code: "const a = window['location'].href.match(/(^[^#]*)/)[0];"},
```

Comment:
Some examples are using `\n` (which I find better than using multiline strings, btw), while other rules are using multiline strings. Can we stick to a single form?